recipes/_double_conversion.rb

```diff
@@ -15,41 +15,50 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-double_conversion_checkout_path = "#{Chef::Config[:file_cache_path]}/double-conversion"
-install_dir = '/usr/local'
-include_file = '/usr/local/lib/libdouble-conversion.a'
 
-directory double_conversion_checkout_path do
-  action :delete
-  recursive true
-  only_if { Dir.exist?(double_conversion_checkout_path) }
-end
+if node['mcrouter']['install_type'] == 'source'
+  double_conversion_checkout_path = "#{Chef::Config[:file_cache_path]}/double-conversion"
+    install_dir = '/usr/local'
+    include_file = '/usr/local/lib/libdouble-conversion.a'
 
-git double_conversion_checkout_path do
-  repository 'https://github.com/google/double-conversion.git'
-  revision node['mcrouter']['double_conversion_commit_hash']
-  action :sync
-  notifies :run, 'execute[Double-conversion - prepare build]'
-end
+    directory double_conversion_checkout_path do
+      action :delete
+      recursive true
+      only_if { Dir.exist?(double_conversion_checkout_path) }
+    end
 
-execute 'Double-conversion - prepare build' do
-  cwd double_conversion_checkout_path
-  command "cmake . -DBUILD_SHARED_LIBS=ON -DCMAKE_INSTALL_PREFIX=#{install_dir}"
-  notifies :run, 'execute[Double-Conversion - Build and install]'
-  not_if { File.exist?(include_file) }
-end
+    git double_conversion_checkout_path do
+      repository 'https://github.com/google/double-conversion.git'
+      revision node['mcrouter']['double_conversion_commit_hash']
+      action :sync
+      notifies :run, 'execute[Double-conversion - prepare build]'
+    end
 
-execute 'Double-Conversion - Build and install' do
-  command 'make && make install'
-  cwd double_conversion_checkout_path
-  not_if { File.exist?(include_file) }
-end
+    execute 'Double-conversion - prepare build' do
+      cwd double_conversion_checkout_path
```

Comment:
This should have `action :nothing` if the intent is to only fire it off when the git checkout syncs.

Reply:
Actually, it's going to sync every time since the dir is being deleted on each run. In that case the `notifies :run` in the `git` resource above is redundant since this `execute` block will always execute next.

Reply:
I removed the `notifies` blocks on the `git` resource and the `execute` resource.

I'm really not sure that the `install from source` code path actually works as I'm only interested in installing from packages :)

Reply:
I almost think the package building should be done from habitat or something like that and chef should just install from packages.....

Reply:
Yeah, that would make more sense.